File: Backend/src/test/java/lan/dk/podcastserver/config/SchedulerConfigTest.java
package lan.dk.podcastserver.config;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.runners.MockitoJUnitRunner;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableScheduling;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Created by kevin on 17/08/15 for Podcast Server
 */
@RunWith(MockitoJUnitRunner.class)
public class SchedulerConfigTest {

    @InjectMocks SchedulerConfig schedulerConfig;

    @Test
    public void should_have_config_injected() {
        assertThat(schedulerConfig).isNotNull();
    }
    
    @Test
    @SuppressWarnings("unchecked")
    public void should_have_spring_annotations() {
        assertThat(schedulerConfig.getClass())
                .hasAnnotations(Configuration.class, EnableScheduling.class);
    }
}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/config/SchedulerConfigTest.kt
package com.github.davinkevin.podcastserver.config

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.junit.jupiter.MockitoExtension
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableScheduling

/**
 * Created by kevin on 17/08/15 for Podcast Server
 */
@ExtendWith(MockitoExtension::class)
class SchedulerConfigTest {

    @InjectMocks lateinit var schedulerConfig: SchedulerConfig

    @Test
    fun should_have_config_injected() {
        assertThat(schedulerConfig).isNotNull
    }

    @Test
    fun should_have_spring_annotations() {
        assertThat(schedulerConfig.javaClass)
                .hasAnnotations(Configuration::class.java, EnableScheduling::class.java)
    }
}
